streamz/tests/test_core.py

```diff
@@ -199,6 +210,21 @@ def test_sliding_window_ref_counts():
         r_prev = r
 
 
+def test_sliding_window_metadata():
+    source = Stream()
+    stream = source.sliding_window(2)
+
+    L = []
+    sink = sink_metadata_to_list(stream, L)
+    stream.connect(sink)
+
+    source.emit(0)
+    source.emit(1, metadata=[{'v': 1}])
+    source.emit(2, metadata=[{'v': 2}])
+    source.emit(3, metadata=[{'v': 3}])
+    assert L == [[{'v': 1}], [{'v': 1}, {'v': 2}], [{'v': 2}, {'v': 3}]]
```

Comment:
I am confused by why this looks like it does. I would have expected
`[{'v': 1}, {'v': 2}, {'v': 3}]`

Reply:
I do not believe that `[{'v': 1}, {'v': 2}, {'v': 3}]` is correct. Note that the above is a list with three lists.  This is how I see the result:

```python
    assert L == \
           [
               [{'v': 1}],            # First emit, because 0 has no metadata 
               [{'v': 1}, {'v': 2}],  # Second emit
               [{'v': 2}, {'v': 3}]   # Third emit
           ]
```

Reply:
Sorry, totally missed the sliding window, you must be right

Reply:
Maybe I can comment on all the tests like above.

Reply:
done.